### PR TITLE
refactor:  rc-input-number refactored with baseInput 

### LINKED ISSUE #42028: InputNumber 的 className 会挂在两个 dom 元素上
### Reproduction link

[Edit on CodeSandbox](https://codesandbox.io/s/zi-ding-yi-zhuang-tai-antd-5-4-5-forked-vlugn7?file=/demo.tsx)

### Steps to reproduce

无

### What is expected?

className 不会同时挂到 div.ant-input-number-affix-wrapper 和 div.ant-input-number

### What is actually happening?

className 同时挂到了 div.ant-input-number-affix-wrapper 和 div.ant-input-number 上

| Environment | Info |
| --- | --- |
| antd | 5.4.5 |
| React | 18.0.0 |
| System | macOS |
| Browser | chrome 110.0.5481.177 |

---
rootClassName 只会挂到根节点上，但是文档上没有说明

### PR DESCRIPTION
**Pull request by MuxinFeng**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [x] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 🔗 Related issue link

https://github.com/react-component/input-number/pull/570

close https://github.com/ant-design/ant-design/issues/42028

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |       rc-input-number is refactored with baseInput      |
| 🇨🇳 Chinese |    rc-input-number 用 baseInput 重构       |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

---

### <samp>🤖 Generated by Copilot at 8a630ff</samp>

Refactored `InputNumber` component to use existing props from `RcInputNumber` for prefix and suffix icons, and fixed some focus and blur issues. Cleaned up unused and redundant code in `components/input-number/index.tsx`.

### <samp>🤖 Generated by Copilot at 8a630ff</samp>

*  Refactor `InputNumber` component to use `RcInputNumber` props for affixes and avoid extra elements ([link](https://github.com/ant-design/ant-design/pull/42762/files?diff=unified&w=0#diff-6c85e13d1b03c66a69812b7717bd3695471bc2a1a381475e085c9dcf28662faaL8), [link](https://github.com/ant-design/ant-design/pull/42762/files?diff=unified&w=0#diff-6c85e13d1b03c66a69812b7717bd3695471bc2a1a381475e085c9dcf28662faaL124-R192))
* Add `mergedBlur` and `mergedFocus` functions to handle `onBlur` and `onFocus` events of `InputNumber` ([link](https://github.com/ant-design/ant-design/pull/42762/files?diff=unified&w=0#diff-6c85e13d1b03c66a69812b7717bd3695471bc2a1a381475e085c9dcf28662faaL113-R122))

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -130,7 +130,7 @@
     "rc-dropdown": "~4.1.0",
     "rc-field-form": "~1.32.0",
     "rc-image": "~5.16.0",
-    "rc-input": "~1.0.4",
+    "rc-input": "^1.1.0",
```
Comment (afc163) [inline package.json:133]:
```suggestion
    "rc-input": "~1.1.0",
```

File: components/input-number/__tests__/__snapshots__/demo.test.tsx.snap [hunk 2]
```diff
@@ -109,7 +109,7 @@ exports[`renders components/input-number/demo/addon.tsx correctly 1`] = `
       class="ant-input-number-group-wrapper"
     >
       <div
-        class="ant-input-number-wrapper ant-input-number-group"
+        class="ant-input-number-wrapper ant-input-number-group ant-input-number-wrapper ant-input-number-group"
```
Comment (afc163) [inline components/input-number/__tests__/__snapshots__/demo.test.tsx.snap:112]: 重复了

File: components/input-number/index.tsx [hunk 3]
```diff
@@ -121,101 +129,64 @@ const InputNumber = React.forwardRef<HTMLInputElement, InputNumberProps>((props,
       prefixCls={prefixCls}
       readOnly={readOnly}
       controls={controlsTemp}
+      onFocus={hasPrefix ? mergedFocus : props.onFocus}
```
Comment (afc163) [inline components/input-number/index.tsx:132]: hasPrefix 可以放到 mergedFocus 里去。

File: components/input-number/index.tsx [hunk 4]
```diff
@@ -121,101 +129,64 @@ const InputNumber = React.forwardRef<HTMLInputElement, InputNumberProps>((props,
...
+      onBlur={hasPrefix ? mergedBlur : props.onBlur}
+      prefix={prefix && prefix}
```
Comment (afc163) [inline components/input-number/index.tsx:134]: 不对。

Comment (afc163) [reply components/input-number/index.tsx:140]: 多了一个 prefix

File: components/input-number/index.tsx [hunk 5]
```diff
@@ -121,101 +135,64 @@ const InputNumber = React.forwardRef<HTMLInputElement, InputNumberProps>((props,
...
+      }
+      classes={{
+        affixWrapper: classNames(
+          getStatusClassNames(`${prefixCls}-affix-wrapper`, mergedStatus, hasFeedback),
+          {
+            [`${prefixCls}-affix-wrapper-focused`]: focused,
+            [`${prefixCls}-affix-wrapper-sm`]: mergedSize === 'small',
+            [`${prefixCls}-affix-wrapper-lg`]: mergedSize === 'large',
+            [`${prefixCls}-affix-wrapper-rtl`]: direction === 'rtl',
+            [`${prefixCls}-affix-wrapper-readonly`]: readOnly,
```
Comment (MadCcc) [inline components/input-number/index.tsx:168]: BaseInput 处理了，可以去掉

File: components/input-number/index.tsx [hunk 6]
```diff
@@ -121,101 +135,64 @@ const InputNumber = React.forwardRef<HTMLInputElement, InputNumberProps>((props,
...
+              {addonBefore}
+            </NoFormStyle>
+          </NoCompactStyle>
+        )
+      }
+      classes={{
+        affixWrapper: classNames(
+          getStatusClassNames(`${prefixCls}-affix-wrapper`, mergedStatus, hasFeedback),
+          {
+            [`${prefixCls}-affix-wrapper-focused`]: focused,
```
Comment (MadCcc) [inline components/input-number/index.tsx:164]: BaseInput 处理

File: components/input-number/index.tsx [hunk 7]
```diff
@@ -121,101 +135,64 @@ const InputNumber = React.forwardRef<HTMLInputElement, InputNumberProps>((props,
...
+        }),
+        group: classNames(
+          {
+            [`${prefixCls}-group-wrapper-sm`]: mergedSize === 'small',
+            [`${prefixCls}-group-wrapper-lg`]: mergedSize === 'large',
+            [`${prefixCls}-group-wrapper-rtl`]: direction === 'rtl',
+          },
+          getStatusClassNames(`${prefixCls}-group-wrapper`, mergedStatus, hasFeedback),
+          hashId,
+          className,
```
Comment (MadCcc) [inline components/input-number/index.tsx:188]: className 也可以交给 BaseInput 处理了，直接传给 rc-input-number 就行

File: components/input-number/index.tsx [hunk 8]
```diff
@@ -121,101 +135,64 @@ const InputNumber = React.forwardRef<HTMLInputElement, InputNumberProps>((props,
...
+          rootClassName,
```
Comment (MadCcc) [inline components/input-number/index.tsx:189]: rootClassName 也可以让 rc 支持下，现在看逻辑和 className 一样的，可以拼起来传给 BaseInput

File: components/input-number/index.tsx [hunk 9]
```diff
@@ -110,8 +109,23 @@ const InputNumber = React.forwardRef<HTMLInputElement, InputNumberProps>((props,
     className,
     !hasPrefix && !hasAddon && rootClassName,
```
Comment (MadCcc) [inline components/input-number/index.tsx:110]: 这边可以去掉了，统一让 BaseInput 处理
https://github.com/ant-design/ant-design/issues/42028

**Review by MadCcc — APPROVED** [review]

**Pull request merged**